File: bytedance/string/043. Multiply Strings.go
func multiply(num1 string, num2 string) string {
    if num1 == "0" || num2 == "0" {
        return "0"
    }
    
    l1, l2 := len(num1), len(num2)
    total := l1 + l2
    res := make([]int, total)
    
    sum := 0
    for i := l1 - 1; i >= 0; i-- {
        for j := l2 - 1; j >= 0; j-- {
            sum = res[i + j + 1] + int(num1[i] - '0') * int(num2[j] - '0')
            res[i + j + 1] = sum % 10
            res[i + j] += sum / 10
        }
    }
    
    re := ""
    for k, v := range res {
        if k == 0 && v == 0 {
            continue
        }
        re += string(v + '0')
    }
    return re
}

File: bytedance/string/043. Multiply Strings.py

#kidding
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        return str(int(num1)*int(num2))


# string
# using int and str just using python's property.
# just like using '1' - '0' and get 1 in C.
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        l1, l2 = len(num1), len(num2)
        total = l1 + l2
        result = [0] * total

        for i in range(l1 - 1, -1, -1):
            for j in range(l2 - 1, -1, -1):
                sum = result[i + j + 1] + int(num1[i]) * int(num2[j])
                result[i + j + 1] = sum % 10
                result[i + j] += sum // 10
        
        index = 0
        while result[index] == 0 and index < total - 1:
            index += 1
        
        re = ""
        while index < total:
            re+=str(result[index])
            index += 1
        return re